### DISCUSSION TITLE
Is it possible to detect clicks on individual particles?

### DISCUSSION THREAD

**Discussion by dillondesilva** [matteobruni/tsparticles#1948] [Q&A] [answered]
Hey there :)

I'm using react-particles-js for a small school project. I was wondering whether it is possible to do onClick events that call a function I've written within my React Component. Extending this a bit further, is it possible to detect when individual particles are clicked rather than the whole canvas (or obtain particle information from the point clicked on the canvas)?

Any help or examples would be much appreciated thank you so much!

**Comment by matteobruni** [comment, marked as answer]
Duplicate: https://github.com/matteobruni/tsparticles/discussions/1610

**Reply by dillondesilva** [reply]
@matteobruni Thanks! I appear to still be having an issue where when I've added a click handler to the container, I can obtain the click events but the clicked particles array returns as empty. My code is as follows:

`particlesLoaded(container) {
        console.log(container);
        container.addClickHandler((event, particles) => this.particleClicked(event, particles))
    }`
`particleClicked(event, particles) {
        console.log(event, particles)
    }`

Are there any conditions under which particles returned could be empty?

**Reply by dillondesilva** [reply]
Some additional information...

This error pops up in console every time I try to click a particle. Perhaps its related to why the clicked particles array is returning empty handed?

<img width="666" alt="Screen Shot 2021-08-06 at 3 39 36 pm" src="https://user-images.githubusercontent.com/10183259/128461716-5185cddd-f0d5-4bb9-b87f-49e0d675aa0a.png">

**Reply by matteobruni** [reply]
Thanks, I'm going to try reproducing the issue later to see if I can get the same error.

**Reply by dillondesilva** [reply]
No worries. Thanks for all the help :) 

If it helps, this is the React Component where I am using `tsparticles`. <https://github.com/dillondesilva/capsicum/blob/main/src/views/CapsuleViewer.js>

**Reply by matteobruni** [reply]
@dillondesilva I have released `1.33.3` with fixes for this feature. Once upgraded it should work fine.

**Reply by dillondesilva** [reply]
@matteobruni Thanks heaps! This is a truly awesome product :)
